src/lib/parseParams.ts

```diff
@@ -0,0 +1,200 @@
+        out: parseAsStringElseUndefined(denoifyParams.out),
+        index: parseAsStringElseUndefined(denoifyParams.index),
+        includes: !Array.isArray(includes)
+            ? undefined
+            : includes.map(elem =>
+                  typeof elem === "string"
+                      ? elem
+                      : {
+                            destDir: parseAsStringElseUndefined(elem.destDir),
+                            destBasename: parseAsStringElseUndefined(elem.destBasename),
+                            src: parseAsStringElseThrow({
+                                param: elem.src,
+                                type: "src in includes array"
+                            })
+                        }
+              ),
+        ports:
+            denoifyParams.ports !== undefined || denoifyParams.ports !== null
+                ? undefined
+                : Object.entries(denoifyParams.ports).reduce(
+                      (prev, [portName, value]) => ({
+                          ...prev,
+                          [portName]: parseAsStringElseThrow({
+                              param: value,
+                              type: "value of ports object"
+                          })
+                      }),
+                      {}
+                  )
+    };
+}
+
+export function configuration() {
+    const packageJson = "package.json";
+    const supportedConfigFile = (() => {
+        const denoify = "denoify";
+        const rcs = ["", ".json", ".yaml", ".yml"];
+        const configs = [".js", ".cjs"];
+        // in order of precedence
+        return [packageJson].concat(
+            rcs.concat(configs).map(extension => `.${denoify}rc${extension}`),
+            configs.map(extension => `${denoify}.config${extension}`)
+        );
+    })();
+
+    function parseConfig({ file, config }: { file: string | undefined; config: string | undefined }): ConfigFileType {
+        if (file === undefined || config === undefined) {
+            return {
+                type: "absent"
+            };
+        }
+        if (file.endsWith(".json")) {
+            return {
+                type: "json",
+                file,
+                config
+            };
+        } else if (file.split(".").length === 2) {
+            return {
+                type: "extensionless",
+                file,
+                config
+            };
+        } else if (file.endsWith(".cjs") || file.endsWith(".js")) {
+            return {
+                type: "js",
+                file,
+                config
+            };
+        } else if (file.endsWith(".yaml") || file.endsWith(".yml")) {
+            return {
+                type: "yaml",
+                file,
+                config
+            };
+        } else {
+            return {
+                type: "absent"
+            };
+        }
+    }
+
+    return {
+        getFileTypeAndContent(content: (file: string) => Promise<string | undefined>) {
```

Comment:
```suggestion
        "getFileTypeAndContent": (content: (file: string) => Promise<string | undefined>) => {
```
`content: (file: string) => Promise<string | undefined>` should be more like `getConfigFileRawContent: (configFileBasename: string)=> Promise<string | unefined>`